File: 1/ruby/b.rb
def fuel_calc(mass)
    #string to int
    mass = mass.to_i
    #divison between ints is floor by deafult
    fuel = mass / 3
    fuel -= 2
    return fuel
end

total_fuel = 0
for line in File.open("input.txt", "r").each_line do
    new_fuel = fuel_calc(line)
    while new_fuel > 0
        total_fuel += new_fuel
        new_fuel = fuel_calc(new_fuel)
    end
#closes file
end
#outputs answer
puts(total_fuel)

File: 1/scala/b.scala
import math.floor
import io.Source.fromFile

object b extends App {
	def fuel_calc(mass: Int): Int ={
    return floor(mass/3).toInt - 2
	}

	var total_fuel : Int = 0
	for (line <- fromFile("input.txt").getLines){
	    var new_fuel: Int = fuel_calc(line.toInt)
	    while (new_fuel >= 0) {
	        total_fuel += new_fuel
	        new_fuel = fuel_calc(new_fuel)
	    }
	}
	println(total_fuel)
}